client/src/pages/Transactions/Transactions.tsx

```diff
@@ -13,26 +15,34 @@ interface Transaction {
 
 const Transactions: React.FC<Transaction> = ({ pageType }) => {
   const [displayAdder, setDisplayAdder] = useState<boolean>(false);
-
   const [rerender, setRerender] = useState<boolean>(false);
+  const { budgetId } = useParams();
 
-  const [incomeItems, setIncomeItems] = useState<Array<ItemData>>([]);
-  const [expenseItems, setExpenseItems] = useState<Array<ItemData>>([]);
+  const fetchTransactions = async () => {
+    const transactions = await getAllItems(budgetId);
+    return transactions;
+  };
 
-  const toggleRerender = () => setRerender(!rerender);
+  const { data, isLoading, status, refetch } = useQuery(
+    "transactions",
+    fetchTransactions
+  );
 
-  const retrieveData = async () => {
-    const expenseData = await getAllItems("expense");
-    const incomeData = await getAllItems("income");
+  const toggleRerender = () => setRerender(!rerender);
 
-    setIncomeItems(incomeData);
-    setExpenseItems(expenseData);
-  };
+  const budgetData = useOutletContext<any>();
```

Comment:
Anything we can use other than ```any```?

Reply:
ah sorry...I was just sick of typescript yelling at me

Reply:
changed this type